[Installation]: Failed to build vLLM from source due to https://github.com/vllm-project/vllm/pull/7174 by bisecting the most recent changes

### Your current environment

```text
The output of `python collect_env.py`
```
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.0
Libc version: glibc-2.31

Python version: 3.10.14 (main, May  6 2024, 19:42:50) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-1045-azure-x86_64-with-glibc2.31
Is CUDA available: N/A
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: NVIDIA A100-SXM4-80GB
GPU 1: NVIDIA A100-SXM4-80GB
GPU 2: NVIDIA A100-SXM4-80GB
GPU 3: NVIDIA A100-SXM4-80GB

Nvidia driver version: 535.86.10
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      48 bits physical, 48 bits virtual
CPU(s):                             96
On-line CPU(s) list:                0-95
Thread(s) per core:                 1
Core(s) per socket:                 48
Socket(s):                          2
NUMA node(s):                       4
Vendor ID:                          AuthenticAMD
CPU family:                         23
Model:                              49
Model name:                         AMD EPYC 7V12 64-Core Processor
Stepping:                           0
CPU MHz:                            2445.443
BogoMIPS:                           4890.88
Hypervisor vendor:                  Microsoft
Virtualization type:                full
L1d cache:                          3 MiB
L1i cache:                          3 MiB
L2 cache:                           48 MiB
L3 cache:                           384 MiB
NUMA node0 CPU(s):                  0-23
NUMA node1 CPU(s):                  24-47
NUMA node2 CPU(s):                  48-71
NUMA node3 CPU(s):                  72-95
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Mitigation; untrained return thunk; SMT disabled
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext perfctr_core ssbd vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 clzero xsaveerptr rdpru arat umip rdpid

Versions of relevant libraries:
[pip3] No relevant packages
[conda] No relevant packages
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: N/A
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0    GPU1    GPU2    GPU3    NIC0    NIC1    NIC2    NIC3    NIC4    NIC5    NIC6    NIC7    NIC8    CPU Affinity  NUMA Affinity    GPU NUMA ID
GPU0     X      NV12    NV12    NV12    SYS     SYS     SYS     SYS     NODE    NODE    SYS     SYS     NODE    0-23  0N/A
GPU1    NV12     X      NV12    NV12    SYS     SYS     SYS     SYS     NODE    NODE    SYS     SYS     NODE    0-23  0N/A
GPU2    NV12    NV12     X      NV12    SYS     SYS     NODE    NODE    SYS     SYS     SYS     SYS     SYS     72-95 3N/A
GPU3    NV12    NV12    NV12     X      SYS     SYS     NODE    NODE    SYS     SYS     SYS     SYS     SYS     72-95 3N/A
NIC0    SYS     SYS     SYS     SYS      X      NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS
NIC1    SYS     SYS     SYS     SYS     NODE     X      SYS     SYS     SYS     SYS     SYS     SYS     SYS
NIC2    SYS     SYS     NODE    NODE    SYS     SYS      X      NODE    SYS     SYS     SYS     SYS     SYS
NIC3    SYS     SYS     NODE    NODE    SYS     SYS     NODE     X      SYS     SYS     SYS     SYS     SYS
NIC4    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS      X      NODE    SYS     SYS     NODE
NIC5    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     NODE     X      SYS     SYS     NODE
NIC6    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      NODE    SYS
NIC7    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE     X      SYS
NIC8    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    SYS     SYS      X

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0
  NIC1: mlx5_1
  NIC2: mlx5_2
  NIC3: mlx5_3
  NIC4: mlx5_4
  NIC5: mlx5_5
  NIC6: mlx5_6
  NIC7: mlx5_7
  NIC8: mlx5_8

### How you are installing vllm

```
conda create -n myenv python=3.10 -y
conda activate myenv 
git clone https://github.com/vllm-project/vllm.git
cd vllm
pip install  --user -e .  # This may take 5-10 minutes.
```

Thanks @simon-mo . Here is the error log:
```
(myenv) aiscuser@node-0:~/vllm$  pip install --user -e . # This may take 5-10 minutes.
Obtaining file:///home/aiscuser/vllm
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Collecting psutil (from vllm==0.5.4+cu118)
  Using cached psutil-6.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (21 kB)
Collecting sentencepiece (from vllm==0.5.4+cu118)
  Using cached sentencepiece-0.2.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.7 kB)
Collecting numpy<2.0.0 (from vllm==0.5.4+cu118)
  Using cached numpy-1.26.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)
Collecting requests (from vllm==0.5.4+cu118)
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting tqdm (from vllm==0.5.4+cu118)
  Using cached tqdm-4.66.5-py3-none-any.whl.metadata (57 kB)
Collecting py-cpuinfo (from vllm==0.5.4+cu118)
  Using cached py_cpuinfo-9.0.0-py3-none-any.whl.metadata (794 bytes)
Collecting transformers>=4.43.2 (from vllm==0.5.4+cu118)
  Using cached transformers-4.44.1-py3-none-any.whl.metadata (43 kB)
Collecting tokenizers>=0.19.1 (from vllm==0.5.4+cu118)
  Using cached tokenizers-0.20.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.7 kB)
Collecting protobuf (from vllm==0.5.4+cu118)
  Using cached protobuf-5.27.3-cp38-abi3-manylinux2014_x86_64.whl.metadata (592 bytes)
Collecting fastapi (from vllm==0.5.4+cu118)
  Using cached fastapi-0.112.1-py3-none-any.whl.metadata (27 kB)
Collecting aiohttp (from vllm==0.5.4+cu118)
  Using cached aiohttp-3.10.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.5 kB)
Collecting openai>=1.0 (from vllm==0.5.4+cu118)
  Using cached openai-1.42.0-py3-none-any.whl.metadata (22 kB)
Collecting uvicorn[standard] (from vllm==0.5.4+cu118)
  Using cached uvicorn-0.30.6-py3-none-any.whl.metadata (6.6 kB)
Collecting pydantic>=2.0 (from vllm==0.5.4+cu118)
  Using cached pydantic-2.8.2-py3-none-any.whl.metadata (125 kB)
Collecting pillow (from vllm==0.5.4+cu118)
  Using cached pillow-10.4.0-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (9.2 kB)
Collecting prometheus-client>=0.18.0 (from vllm==0.5.4+cu118)
  Using cached prometheus_client-0.20.0-py3-none-any.whl.metadata (1.8 kB)
Collecting prometheus-fastapi-instrumentator>=7.0.0 (from vllm==0.5.4+cu118)
  Using cached prometheus_fastapi_instrumentator-7.0.0-py3-none-any.whl.metadata (13 kB)
Collecting tiktoken>=0.6.0 (from vllm==0.5.4+cu118)
  Using cached tiktoken-0.7.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.6 kB)
Collecting lm-format-enforcer==0.10.6 (from vllm==0.5.4+cu118)
  Using cached lm_format_enforcer-0.10.6-py3-none-any.whl.metadata (16 kB)
Collecting outlines<0.1,>=0.0.43 (from vllm==0.5.4+cu118)
  Using cached outlines-0.0.46-py3-none-any.whl.metadata (15 kB)
Collecting typing-extensions>=4.10 (from vllm==0.5.4+cu118)
  Using cached typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting filelock>=3.10.4 (from vllm==0.5.4+cu118)
  Using cached filelock-3.15.4-py3-none-any.whl.metadata (2.9 kB)
Collecting pyzmq (from vllm==0.5.4+cu118)
  Using cached pyzmq-26.1.1-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (6.2 kB)
Collecting msgspec (from vllm==0.5.4+cu118)
  Using cached msgspec-0.18.6-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.9 kB)
Collecting librosa (from vllm==0.5.4+cu118)
  Using cached librosa-0.10.2.post1-py3-none-any.whl.metadata (8.6 kB)
Collecting soundfile (from vllm==0.5.4+cu118)
  Using cached soundfile-0.12.1-py2.py3-none-manylinux_2_31_x86_64.whl.metadata (14 kB)
Collecting gguf==0.9.1 (from vllm==0.5.4+cu118)
  Using cached gguf-0.9.1-py3-none-any.whl.metadata (3.3 kB)
Collecting importlib-metadata (from vllm==0.5.4+cu118)
  Using cached importlib_metadata-8.4.0-py3-none-any.whl.metadata (4.7 kB)
Collecting ray>=2.9 (from vllm==0.5.4+cu118)
  Using cached ray-2.34.0-cp310-cp310-manylinux2014_x86_64.whl.metadata (13 kB)
Collecting nvidia-ml-py (from vllm==0.5.4+cu118)
  Using cached nvidia_ml_py-12.560.30-py3-none-any.whl.metadata (8.6 kB)
Collecting torch==2.4.0 (from vllm==0.5.4+cu118)
  Using cached torch-2.4.0-cp310-cp310-manylinux1_x86_64.whl.metadata (26 kB)
Collecting torchvision==0.19 (from vllm==0.5.4+cu118)
  Using cached torchvision-0.19.0-cp310-cp310-manylinux1_x86_64.whl.metadata (6.0 kB)
Collecting xformers==0.0.27.post2 (from vllm==0.5.4+cu118)
  Using cached xformers-0.0.27.post2-cp310-cp310-manylinux2014_x86_64.whl.metadata (1.0 kB)
Collecting vllm-flash-attn==2.6.1 (from vllm==0.5.4+cu118)
  Using cached vllm_flash_attn-2.6.1-cp310-cp310-manylinux1_x86_64.whl.metadata (476 bytes)
Collecting interegular>=0.3.2 (from lm-format-enforcer==0.10.6->vllm==0.5.4+cu118)
  Using cached interegular-0.3.3-py37-none-any.whl.metadata (3.0 kB)
Collecting packaging (from lm-format-enforcer==0.10.6->vllm==0.5.4+cu118)
  Using cached packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
Collecting pyyaml (from lm-format-enforcer==0.10.6->vllm==0.5.4+cu118)
  Using cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting sympy (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached sympy-1.13.2-py3-none-any.whl.metadata (12 kB)
Collecting networkx (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached networkx-3.3-py3-none-any.whl.metadata (5.1 kB)
Collecting jinja2 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting fsspec (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached fsspec-2024.6.1-py3-none-any.whl.metadata (11 kB)
Collecting nvidia-cuda-nvrtc-cu12==12.1.105 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)
Collecting nvidia-cuda-runtime-cu12==12.1.105 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)
Collecting nvidia-cuda-cupti-cu12==12.1.105 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cuda_cupti_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.6 kB)
Collecting nvidia-cudnn-cu12==9.1.0.70 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cudnn_cu12-9.1.0.70-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)
Collecting nvidia-cublas-cu12==12.1.3.1 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cublas_cu12-12.1.3.1-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)
Collecting nvidia-cufft-cu12==11.0.2.54 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cufft_cu12-11.0.2.54-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)
Collecting nvidia-curand-cu12==10.3.2.106 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_curand_cu12-10.3.2.106-py3-none-manylinux1_x86_64.whl.metadata (1.5 kB)
Collecting nvidia-cusolver-cu12==11.4.5.107 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cusolver_cu12-11.4.5.107-py3-none-manylinux1_x86_64.whl.metadata (1.6 kB)
Collecting nvidia-cusparse-cu12==12.1.0.106 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_cusparse_cu12-12.1.0.106-py3-none-manylinux1_x86_64.whl.metadata (1.6 kB)
Collecting nvidia-nccl-cu12==2.20.5 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_nccl_cu12-2.20.5-py3-none-manylinux2014_x86_64.whl.metadata (1.8 kB)
Collecting nvidia-nvtx-cu12==12.1.105 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_nvtx_cu12-12.1.105-py3-none-manylinux1_x86_64.whl.metadata (1.7 kB)
Collecting triton==3.0.0 (from torch==2.4.0->vllm==0.5.4+cu118)
  Using cached triton-3.0.0-1-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl.metadata (1.3 kB)
Collecting nvidia-nvjitlink-cu12 (from nvidia-cusolver-cu12==11.4.5.107->torch==2.4.0->vllm==0.5.4+cu118)
  Using cached nvidia_nvjitlink_cu12-12.6.20-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)
Collecting anyio<5,>=3.5.0 (from openai>=1.0->vllm==0.5.4+cu118)
  Using cached anyio-4.4.0-py3-none-any.whl.metadata (4.6 kB)
Collecting distro<2,>=1.7.0 (from openai>=1.0->vllm==0.5.4+cu118)
  Using cached distro-1.9.0-py3-none-any.whl.metadata (6.8 kB)
Collecting httpx<1,>=0.23.0 (from openai>=1.0->vllm==0.5.4+cu118)
  Using cached httpx-0.27.0-py3-none-any.whl.metadata (7.2 kB)
Collecting jiter<1,>=0.4.0 (from openai>=1.0->vllm==0.5.4+cu118)
  Using cached jiter-0.5.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.6 kB)
Collecting sniffio (from openai>=1.0->vllm==0.5.4+cu118)
  Using cached sniffio-1.3.1-py3-none-any.whl.metadata (3.9 kB)
Collecting lark (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached lark-1.2.2-py3-none-any.whl.metadata (1.8 kB)
Collecting nest-asyncio (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached nest_asyncio-1.6.0-py3-none-any.whl.metadata (2.8 kB)
Collecting cloudpickle (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached cloudpickle-3.0.0-py3-none-any.whl.metadata (7.0 kB)
Collecting diskcache (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached diskcache-5.6.3-py3-none-any.whl.metadata (20 kB)
Collecting numba (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached numba-0.60.0-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl.metadata (2.7 kB)
Collecting referencing (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached referencing-0.35.1-py3-none-any.whl.metadata (2.8 kB)
Collecting jsonschema (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached jsonschema-4.23.0-py3-none-any.whl.metadata (7.9 kB)
Collecting datasets (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached datasets-2.21.0-py3-none-any.whl.metadata (21 kB)
Collecting pycountry (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached pycountry-24.6.1-py3-none-any.whl.metadata (12 kB)
Collecting pyairports (from outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached pyairports-2.1.1-py3-none-any.whl.metadata (1.7 kB)
Collecting starlette<1.0.0,>=0.30.0 (from prometheus-fastapi-instrumentator>=7.0.0->vllm==0.5.4+cu118)
  Using cached starlette-0.38.2-py3-none-any.whl.metadata (5.9 kB)
Collecting annotated-types>=0.4.0 (from pydantic>=2.0->vllm==0.5.4+cu118)
  Using cached annotated_types-0.7.0-py3-none-any.whl.metadata (15 kB)
Collecting pydantic-core==2.20.1 (from pydantic>=2.0->vllm==0.5.4+cu118)
  Using cached pydantic_core-2.20.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.6 kB)
Collecting click>=7.0 (from ray>=2.9->vllm==0.5.4+cu118)
  Using cached click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Collecting msgpack<2.0.0,>=1.0.0 (from ray>=2.9->vllm==0.5.4+cu118)
  Using cached msgpack-1.0.8-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.1 kB)
Collecting aiosignal (from ray>=2.9->vllm==0.5.4+cu118)
  Using cached aiosignal-1.3.1-py3-none-any.whl.metadata (4.0 kB)
Collecting frozenlist (from ray>=2.9->vllm==0.5.4+cu118)
  Using cached frozenlist-1.4.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)
Collecting regex>=2022.1.18 (from tiktoken>=0.6.0->vllm==0.5.4+cu118)
  Using cached regex-2024.7.24-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (40 kB)
Collecting charset-normalizer<4,>=2 (from requests->vllm==0.5.4+cu118)
  Using cached charset_normalizer-3.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (33 kB)
Collecting idna<4,>=2.5 (from requests->vllm==0.5.4+cu118)
  Using cached idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Collecting urllib3<3,>=1.21.1 (from requests->vllm==0.5.4+cu118)
  Using cached urllib3-2.2.2-py3-none-any.whl.metadata (6.4 kB)
Collecting certifi>=2017.4.17 (from requests->vllm==0.5.4+cu118)
  Using cached certifi-2024.7.4-py3-none-any.whl.metadata (2.2 kB)
Collecting huggingface-hub<1.0,>=0.16.4 (from tokenizers>=0.19.1->vllm==0.5.4+cu118)
  Using cached huggingface_hub-0.24.6-py3-none-any.whl.metadata (13 kB)
Collecting safetensors>=0.4.1 (from transformers>=4.43.2->vllm==0.5.4+cu118)
  Using cached safetensors-0.4.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.8 kB)
Collecting tokenizers>=0.19.1 (from vllm==0.5.4+cu118)
  Using cached tokenizers-0.19.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.7 kB)
Collecting aiohappyeyeballs>=2.3.0 (from aiohttp->vllm==0.5.4+cu118)
  Using cached aiohappyeyeballs-2.4.0-py3-none-any.whl.metadata (5.9 kB)
Collecting attrs>=17.3.0 (from aiohttp->vllm==0.5.4+cu118)
  Using cached attrs-24.2.0-py3-none-any.whl.metadata (11 kB)
Collecting multidict<7.0,>=4.5 (from aiohttp->vllm==0.5.4+cu118)
  Using cached multidict-6.0.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.2 kB)
Collecting yarl<2.0,>=1.0 (from aiohttp->vllm==0.5.4+cu118)
  Using cached yarl-1.9.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (31 kB)
Collecting async-timeout<5.0,>=4.0 (from aiohttp->vllm==0.5.4+cu118)
  Using cached async_timeout-4.0.3-py3-none-any.whl.metadata (4.2 kB)
Collecting zipp>=0.5 (from importlib-metadata->vllm==0.5.4+cu118)
  Using cached zipp-3.20.0-py3-none-any.whl.metadata (3.6 kB)
Collecting audioread>=2.1.9 (from librosa->vllm==0.5.4+cu118)
  Using cached audioread-3.0.1-py3-none-any.whl.metadata (8.4 kB)
Collecting scipy>=1.2.0 (from librosa->vllm==0.5.4+cu118)
  Using cached scipy-1.14.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)
Collecting scikit-learn>=0.20.0 (from librosa->vllm==0.5.4+cu118)
  Using cached scikit_learn-1.5.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)
Collecting joblib>=0.14 (from librosa->vllm==0.5.4+cu118)
  Using cached joblib-1.4.2-py3-none-any.whl.metadata (5.4 kB)
Collecting decorator>=4.3.0 (from librosa->vllm==0.5.4+cu118)
  Using cached decorator-5.1.1-py3-none-any.whl.metadata (4.0 kB)
Collecting pooch>=1.1 (from librosa->vllm==0.5.4+cu118)
  Using cached pooch-1.8.2-py3-none-any.whl.metadata (10 kB)
Collecting soxr>=0.3.2 (from librosa->vllm==0.5.4+cu118)
  Using cached soxr-0.4.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.5 kB)
Collecting lazy-loader>=0.1 (from librosa->vllm==0.5.4+cu118)
  Using cached lazy_loader-0.4-py3-none-any.whl.metadata (7.6 kB)
Collecting cffi>=1.0 (from soundfile->vllm==0.5.4+cu118)
  Using cached cffi-1.17.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.5 kB)
Collecting h11>=0.8 (from uvicorn[standard]->vllm==0.5.4+cu118)
  Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
Collecting httptools>=0.5.0 (from uvicorn[standard]->vllm==0.5.4+cu118)
  Using cached httptools-0.6.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.6 kB)
Collecting python-dotenv>=0.13 (from uvicorn[standard]->vllm==0.5.4+cu118)
  Using cached python_dotenv-1.0.1-py3-none-any.whl.metadata (23 kB)
Collecting uvloop!=0.15.0,!=0.15.1,>=0.14.0 (from uvicorn[standard]->vllm==0.5.4+cu118)
  Using cached uvloop-0.20.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)
Collecting watchfiles>=0.13 (from uvicorn[standard]->vllm==0.5.4+cu118)
  Using cached watchfiles-0.23.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)
Collecting websockets>=10.4 (from uvicorn[standard]->vllm==0.5.4+cu118)
  Using cached websockets-13.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.7 kB)
Collecting exceptiongroup>=1.0.2 (from anyio<5,>=3.5.0->openai>=1.0->vllm==0.5.4+cu118)
  Using cached exceptiongroup-1.2.2-py3-none-any.whl.metadata (6.6 kB)
Collecting pycparser (from cffi>=1.0->soundfile->vllm==0.5.4+cu118)
  Using cached pycparser-2.22-py3-none-any.whl.metadata (943 bytes)
Collecting httpcore==1.* (from httpx<1,>=0.23.0->openai>=1.0->vllm==0.5.4+cu118)
  Using cached httpcore-1.0.5-py3-none-any.whl.metadata (20 kB)
Collecting llvmlite<0.44,>=0.43.0dev0 (from numba->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached llvmlite-0.43.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.8 kB)
Collecting platformdirs>=2.5.0 (from pooch>=1.1->librosa->vllm==0.5.4+cu118)
  Using cached platformdirs-4.2.2-py3-none-any.whl.metadata (11 kB)
Collecting threadpoolctl>=3.1.0 (from scikit-learn>=0.20.0->librosa->vllm==0.5.4+cu118)
  Using cached threadpoolctl-3.5.0-py3-none-any.whl.metadata (13 kB)
Collecting pyarrow>=15.0.0 (from datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached pyarrow-17.0.0-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (3.3 kB)
Collecting dill<0.3.9,>=0.3.0 (from datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached dill-0.3.8-py3-none-any.whl.metadata (10 kB)
Collecting pandas (from datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached pandas-2.2.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (19 kB)
Collecting xxhash (from datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached xxhash-3.5.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)
Collecting multiprocess (from datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached multiprocess-0.70.16-py310-none-any.whl.metadata (7.2 kB)
Collecting MarkupSafe>=2.0 (from jinja2->torch==2.4.0->vllm==0.5.4+cu118)
  Using cached MarkupSafe-2.1.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)
Collecting jsonschema-specifications>=2023.03.6 (from jsonschema->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached jsonschema_specifications-2023.12.1-py3-none-any.whl.metadata (3.0 kB)
Collecting rpds-py>=0.7.1 (from jsonschema->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached rpds_py-0.20.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.2 kB)
Collecting mpmath<1.4,>=1.1.0 (from sympy->torch==2.4.0->vllm==0.5.4+cu118)
  Using cached mpmath-1.3.0-py3-none-any.whl.metadata (8.6 kB)
Collecting python-dateutil>=2.8.2 (from pandas->datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting pytz>=2020.1 (from pandas->datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached pytz-2024.1-py2.py3-none-any.whl.metadata (22 kB)
Collecting tzdata>=2022.7 (from pandas->datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting six>=1.5 (from python-dateutil>=2.8.2->pandas->datasets->outlines<0.1,>=0.0.43->vllm==0.5.4+cu118)
  Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Using cached gguf-0.9.1-py3-none-any.whl (49 kB)
Using cached lm_format_enforcer-0.10.6-py3-none-any.whl (43 kB)
Using cached torch-2.4.0-cp310-cp310-manylinux1_x86_64.whl (797.2 MB)
Using cached torchvision-0.19.0-cp310-cp310-manylinux1_x86_64.whl (7.0 MB)
Using cached vllm_flash_attn-2.6.1-cp310-cp310-manylinux1_x86_64.whl (75.9 MB)
Using cached xformers-0.0.27.post2-cp310-cp310-manylinux2014_x86_64.whl (20.8 MB)
Using cached nvidia_cublas_cu12-12.1.3.1-py3-none-manylinux1_x86_64.whl (410.6 MB)
Using cached nvidia_cuda_cupti_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (14.1 MB)
Using cached nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (23.7 MB)
Using cached nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (823 kB)
Using cached nvidia_cudnn_cu12-9.1.0.70-py3-none-manylinux2014_x86_64.whl (664.8 MB)
Using cached nvidia_cufft_cu12-11.0.2.54-py3-none-manylinux1_x86_64.whl (121.6 MB)
Using cached nvidia_curand_cu12-10.3.2.106-py3-none-manylinux1_x86_64.whl (56.5 MB)
Using cached nvidia_cusolver_cu12-11.4.5.107-py3-none-manylinux1_x86_64.whl (124.2 MB)
Using cached nvidia_cusparse_cu12-12.1.0.106-py3-none-manylinux1_x86_64.whl (196.0 MB)
Using cached nvidia_nccl_cu12-2.20.5-py3-none-manylinux2014_x86_64.whl (176.2 MB)
Using cached nvidia_nvtx_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (99 kB)
Using cached triton-3.0.0-1-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (209.4 MB)
Using cached filelock-3.15.4-py3-none-any.whl (16 kB)
Using cached numpy-1.26.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)
Using cached openai-1.42.0-py3-none-any.whl (362 kB)
Using cached outlines-0.0.46-py3-none-any.whl (101 kB)
Using cached pillow-10.4.0-cp310-cp310-manylinux_2_28_x86_64.whl (4.5 MB)
Using cached prometheus_client-0.20.0-py3-none-any.whl (54 kB)
Using cached prometheus_fastapi_instrumentator-7.0.0-py3-none-any.whl (19 kB)
Using cached pydantic-2.8.2-py3-none-any.whl (423 kB)
Using cached pydantic_core-2.20.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.1 MB)
Using cached ray-2.34.0-cp310-cp310-manylinux2014_x86_64.whl (64.9 MB)
Using cached protobuf-5.27.3-cp38-abi3-manylinux2014_x86_64.whl (309 kB)
Using cached tiktoken-0.7.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Using cached tqdm-4.66.5-py3-none-any.whl (78 kB)
Using cached transformers-4.44.1-py3-none-any.whl (9.5 MB)
Using cached tokenizers-0.19.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.6 MB)
Using cached typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Using cached aiohttp-3.10.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.2 MB)
Using cached fastapi-0.112.1-py3-none-any.whl (93 kB)
Using cached importlib_metadata-8.4.0-py3-none-any.whl (26 kB)
Using cached librosa-0.10.2.post1-py3-none-any.whl (260 kB)
Using cached soundfile-0.12.1-py2.py3-none-manylinux_2_31_x86_64.whl (1.2 MB)
Using cached msgspec-0.18.6-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (210 kB)
Using cached nvidia_ml_py-12.560.30-py3-none-any.whl (40 kB)
Using cached psutil-6.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (290 kB)
Using cached py_cpuinfo-9.0.0-py3-none-any.whl (22 kB)
Using cached pyzmq-26.1.1-cp310-cp310-manylinux_2_28_x86_64.whl (868 kB)
Using cached sentencepiece-0.2.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)
Using cached aiohappyeyeballs-2.4.0-py3-none-any.whl (12 kB)
Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Using cached annotated_types-0.7.0-py3-none-any.whl (13 kB)
Using cached anyio-4.4.0-py3-none-any.whl (86 kB)
Using cached async_timeout-4.0.3-py3-none-any.whl (5.7 kB)
Using cached attrs-24.2.0-py3-none-any.whl (63 kB)
Using cached audioread-3.0.1-py3-none-any.whl (23 kB)
Using cached certifi-2024.7.4-py3-none-any.whl (162 kB)
Using cached cffi-1.17.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (445 kB)
Using cached charset_normalizer-3.3.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (142 kB)
Using cached click-8.1.7-py3-none-any.whl (97 kB)
Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
Using cached distro-1.9.0-py3-none-any.whl (20 kB)
Using cached frozenlist-1.4.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (239 kB)
Using cached h11-0.14.0-py3-none-any.whl (58 kB)
Using cached httptools-0.6.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (341 kB)
Using cached httpx-0.27.0-py3-none-any.whl (75 kB)
Using cached httpcore-1.0.5-py3-none-any.whl (77 kB)
Using cached huggingface_hub-0.24.6-py3-none-any.whl (417 kB)
Using cached fsspec-2024.6.1-py3-none-any.whl (177 kB)
Using cached idna-3.7-py3-none-any.whl (66 kB)
Using cached interegular-0.3.3-py37-none-any.whl (23 kB)
Using cached jiter-0.5.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (318 kB)
Using cached joblib-1.4.2-py3-none-any.whl (301 kB)
Using cached lazy_loader-0.4-py3-none-any.whl (12 kB)
Using cached msgpack-1.0.8-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (385 kB)
Using cached multidict-6.0.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (124 kB)
Using cached numba-0.60.0-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (3.7 MB)
Using cached packaging-24.1-py3-none-any.whl (53 kB)
Using cached pooch-1.8.2-py3-none-any.whl (64 kB)
Using cached python_dotenv-1.0.1-py3-none-any.whl (19 kB)
Using cached PyYAML-6.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (751 kB)
Using cached regex-2024.7.24-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (776 kB)
Using cached safetensors-0.4.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (435 kB)
Using cached scikit_learn-1.5.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.4 MB)
Using cached scipy-1.14.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (41.2 MB)
Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
Using cached soxr-0.4.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)
Using cached starlette-0.38.2-py3-none-any.whl (72 kB)
Using cached urllib3-2.2.2-py3-none-any.whl (121 kB)
Using cached uvloop-0.20.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.4 MB)
Using cached watchfiles-0.23.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (427 kB)
Using cached websockets-13.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (157 kB)
Using cached yarl-1.9.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (301 kB)
Using cached zipp-3.20.0-py3-none-any.whl (9.4 kB)
Using cached cloudpickle-3.0.0-py3-none-any.whl (20 kB)
Using cached datasets-2.21.0-py3-none-any.whl (527 kB)
Using cached diskcache-5.6.3-py3-none-any.whl (45 kB)
Using cached jinja2-3.1.4-py3-none-any.whl (133 kB)
Using cached jsonschema-4.23.0-py3-none-any.whl (88 kB)
Using cached referencing-0.35.1-py3-none-any.whl (26 kB)
Using cached lark-1.2.2-py3-none-any.whl (111 kB)
Using cached nest_asyncio-1.6.0-py3-none-any.whl (5.2 kB)
Using cached networkx-3.3-py3-none-any.whl (1.7 MB)
Using cached pyairports-2.1.1-py3-none-any.whl (371 kB)
Using cached pycountry-24.6.1-py3-none-any.whl (6.3 MB)
Using cached sympy-1.13.2-py3-none-any.whl (6.2 MB)
Using cached uvicorn-0.30.6-py3-none-any.whl (62 kB)
Using cached dill-0.3.8-py3-none-any.whl (116 kB)
Using cached exceptiongroup-1.2.2-py3-none-any.whl (16 kB)
Using cached jsonschema_specifications-2023.12.1-py3-none-any.whl (18 kB)
Using cached llvmlite-0.43.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (43.9 MB)
Using cached MarkupSafe-2.1.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Using cached platformdirs-4.2.2-py3-none-any.whl (18 kB)
Using cached pyarrow-17.0.0-cp310-cp310-manylinux_2_28_x86_64.whl (39.9 MB)
Using cached rpds_py-0.20.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (354 kB)
Using cached threadpoolctl-3.5.0-py3-none-any.whl (18 kB)
Using cached multiprocess-0.70.16-py310-none-any.whl (134 kB)
Using cached nvidia_nvjitlink_cu12-12.6.20-py3-none-manylinux2014_x86_64.whl (19.7 MB)
Using cached pandas-2.2.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.0 MB)
Using cached pycparser-2.22-py3-none-any.whl (117 kB)
Using cached xxhash-3.5.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (194 kB)
Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Using cached pytz-2024.1-py2.py3-none-any.whl (505 kB)
Using cached tzdata-2024.1-py2.py3-none-any.whl (345 kB)
Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Building wheels for collected packages: vllm
  Building editable for vllm (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building editable for vllm (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [192 lines of output]
      /tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/torch/_subclasses/functional_tensor.py:258: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:84.)
        cpu = _conversion_method_template(device=torch.device("cpu"))
      running editable_wheel
      creating /tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info
      writing /tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/dependency_links.txt
      writing entry points to /tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/entry_points.txt
      writing requirements to /tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/requires.txt
      writing top-level names to /tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/top_level.txt
      writing manifest file '/tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/SOURCES.txt'
      reading manifest file '/tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      writing manifest file '/tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm.egg-info/SOURCES.txt'
      creating '/tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm-0.5.4+cu118.dist-info'
      creating /tmp/pip-wheel-khv6cnct/.tmp-v8tzgcjn/vllm-0.5.4+cu118.dist-info/WHEEL
      running build_py
      running build_ext
      Using MAX_JOBS=256 as the number of jobs.
      Using NVCC_THREADS=16 as the number of nvcc threads.
      -- The CXX compiler identification is GNU 9.4.0
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Build type: RelWithDebInfo
      -- Target device: cuda
      -- Found Python: /home/aiscuser/.conda/envs/myenv/bin/python (found version "3.10.14") found components: Interpreter Development.Module Development.SABIModule
      -- Found python matching: /home/aiscuser/.conda/envs/myenv/bin/python.
      -- Found CUDA: /usr/local/cuda (found version "11.8")
      -- The CUDA compiler identification is NVIDIA 11.8.89
      -- Detecting CUDA compiler ABI info
      -- Detecting CUDA compiler ABI info - done
      -- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc - skipped
      -- Detecting CUDA compile features
      -- Detecting CUDA compile features - done
      -- Found CUDAToolkit: /usr/local/cuda/include (found version "11.8.89")
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
      -- Looking for pthread_create in pthreads
      -- Looking for pthread_create in pthreads - not found
      -- Looking for pthread_create in pthread
      -- Looking for pthread_create in pthread - found
      -- Found Threads: TRUE
      -- Caffe2: CUDA detected: 11.8
      -- Caffe2: CUDA nvcc is: /usr/local/cuda/bin/nvcc
      -- Caffe2: CUDA toolkit directory: /usr/local/cuda
      -- Caffe2: Header version is: 11.8
      -- /usr/local/cuda/lib64/libnvrtc.so shorthash is 672ee683
      -- USE_CUDNN is set to 0. Compiling without cuDNN support
      -- USE_CUSPARSELT is set to 0. Compiling without cuSPARSELt support
      -- Autodetected CUDA architecture(s):  8.0 8.0 8.0 8.0
      -- Added CUDA NVCC flags for: -gencode;arch=compute_80,code=sm_80
      CMake Warning at /tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/torch/share/cmake/Torch/TorchConfig.cmake:22 (message):
        static library kineto_LIBRARY-NOTFOUND not found.
      Call Stack (most recent call first):
        /tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/torch/share/cmake/Torch/TorchConfig.cmake:120 (append_torchlib_if_found)
        CMakeLists.txt:67 (find_package)


      -- Found Torch: /tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/torch/lib/libtorch.so
      -- Enabling core extension.
      -- CUDA supported arches: 7.0;7.5;8.0;8.6;8.9;9.0
      -- CUDA target arches: 80-real
      -- CMake Version: 3.30.2
      -- CUTLASS 3.5.1
      -- CUDART: /usr/local/cuda/lib64/libcudart.so
      -- CUDA Driver: /usr/local/cuda/lib64/stubs/libcuda.so
      -- NVRTC: /usr/local/cuda/lib64/libnvrtc.so
      -- Default Install Location: install
      -- Found Python3: /home/aiscuser/.conda/envs/myenv/bin/python3.10 (found suitable version "3.10.14", minimum required is "3.5") found components: Interpreter
      -- Make cute::tuple be the new standard-layout tuple type
      -- CUDA Compilation Architectures: 70;72;75;80;86;87;89;90
      -- Enable caching of reference results in conv unit tests
      -- Enable rigorous conv problem sizes in conv unit tests
      -- Using NVCC flags: --expt-relaxed-constexpr;-DCUTE_USE_PACKED_TUPLE=1;-DCUTLASS_TEST_LEVEL=0;-DCUTLASS_TEST_ENABLE_CACHED_RESULTS=1;-DCUTLASS_CONV_UNIT_TEST_RIGOROUS_SIZE_ENABLED=1;-DCUTLASS_DEBUG_TRACE_LEVEL=0;-Xcompiler=-Wconversion;-Xcompiler=-fno-strict-aliasing;-lineinfo
      fatal: not a git repository (or any of the parent directories): .git
      -- CUTLASS Revision: Unable to detect, Git returned code 128.
      -- Configuring cublas ...
      -- cuBLAS Disabled.
      -- Configuring cuBLAS ... done.
      CMake Error at CMakeLists.txt:245 (message):
        Machete generation failed.  Result: "1"

        Check the log for details:
        /tmp/tmp7h5m3uxb.build-temp/machete_generation.log


      -- Configuring incomplete, errors occurred!
      Traceback (most recent call last):
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 135, in run
          self._create_wheel_file(bdist_wheel)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 338, in _create_wheel_file
          files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 261, in _run_build_commands
          self._run_build_subcommands()
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 288, in _run_build_subcommands
          self.run_command(name)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 948, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 983, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 96, in run
          _build_ext.run(self)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
          self.build_extensions()
        File "<string>", line 222, in build_extensions
        File "<string>", line 204, in configure
        File "/home/aiscuser/.conda/envs/myenv/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '/home/aiscuser/vllm', '-G', 'Ninja', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/tmp7v3db2j6.build-lib/vllm', '-DCMAKE_ARCHIVE_OUTPUT_DIRECTORY=/tmp/tmp7h5m3uxb.build-temp', '-DVLLM_TARGET_DEVICE=cuda', '-DVLLM_PYTHON_EXECUTABLE=/home/aiscuser/.conda/envs/myenv/bin/python', '-DNVCC_THREADS=16', '-DCMAKE_JOB_POOL_COMPILE:STRING=compile', '-DCMAKE_JOB_POOLS:STRING=compile=16']' returned non-zero exit status 1.
      /tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py:983: _DebuggingTips: Problem in editable installation.
      !!

              ********************************************************************************
              An error happened while installing `vllm` in editable mode.

              The following steps are recommended to help debug this problem:

              - Try to install the project normally, without using the editable mode.
                Does the error still persist?
                (If it does, try fixing the problem before attempting the editable mode).
              - If you are using binary extensions, make sure you have all OS-level
                dependencies installed (e.g. compilers, toolchains, binary libraries, ...).
              - Try the latest version of setuptools (maybe the error was already fixed).
              - If you (or your project dependencies) are using any setuptools extension
                or customization, make sure they support the editable mode.

              After following the steps above, if the problem still persists and
              you think this is related to how setuptools handles editable installations,
              please submit a reproducible example
              (see https://stackoverflow.com/help/minimal-reproducible-example) to:

                  https://github.com/pypa/setuptools/issues

              See https://setuptools.pypa.io/en/latest/userguide/development_mode.html for details.
              ********************************************************************************

      !!
        cmd_obj.run()
      Traceback (most recent call last):
        File "/home/aiscuser/.conda/envs/myenv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/aiscuser/.conda/envs/myenv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/aiscuser/.conda/envs/myenv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 273, in build_editable
          return hook(wheel_directory, config_settings, metadata_directory)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 458, in build_editable
          return self._build_with_temp_dir(
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 402, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 318, in run_setup
          exec(code, locals())
        File "<string>", line 470, in <module>
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 111, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 184, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 964, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 948, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 983, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 135, in run
          self._create_wheel_file(bdist_wheel)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 338, in _create_wheel_file
          files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 261, in _run_build_commands
          self._run_build_subcommands()
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 288, in _run_build_subcommands
          self.run_command(name)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 948, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 983, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/command/build_ext.py", line 96, in run
          _build_ext.run(self)
        File "/tmp/pip-build-env-k21yn32m/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
          self.build_extensions()
        File "<string>", line 222, in build_extensions
        File "<string>", line 204, in configure
        File "/home/aiscuser/.conda/envs/myenv/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '/home/aiscuser/vllm', '-G', 'Ninja', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/tmp7v3db2j6.build-lib/vllm', '-DCMAKE_ARCHIVE_OUTPUT_DIRECTORY=/tmp/tmp7h5m3uxb.build-temp', '-DVLLM_TARGET_DEVICE=cuda', '-DVLLM_PYTHON_EXECUTABLE=/home/aiscuser/.conda/envs/myenv/bin/python', '-DNVCC_THREADS=16', '-DCMAKE_JOB_POOL_COMPILE:STRING=compile', '-DCMAKE_JOB_POOLS:STRING=compile=16']' returned non-zero exit status 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for vllm
Failed to build vllm
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (vllm)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,35 +228,38 @@ if(VLLM_GPU_LANG STREQUAL "CUDA")
   endif()
 
   #
-  # For the Machete kernels we automatically generate sources for various 
-  # preselected input type pairs and schedules.
-  # Generate sources:
-  execute_process(
-    COMMAND ${CMAKE_COMMAND} -E env 
-    PYTHONPATH=${CMAKE_CURRENT_SOURCE_DIR}/csrc/cutlass_extensions/:${CUTLASS_DIR}/python/:${VLLM_PYTHON_PATH}:$PYTHONPATH 
-      ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/csrc/quantization/machete/generate.py
-    RESULT_VARIABLE machete_generation_result
-    OUTPUT_VARIABLE machete_generation_output
-    OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/machete_generation.log
-    ERROR_FILE ${CMAKE_CURRENT_BINARY_DIR}/machete_generation.log
-  )
+  # Machete kernels
 
-  if (NOT machete_generation_result EQUAL 0)
-    message(FATAL_ERROR "Machete generation failed."
-                        " Result: \"${machete_generation_result}\"" 
-                        "\nCheck the log for details: "
-                        "${CMAKE_CURRENT_BINARY_DIR}/machete_generation.log")
-  else()
-    message(STATUS "Machete generation completed successfully.")
-  endif()
-
-  # Add machete generated sources
-  file(GLOB MACHETE_GEN_SOURCES "csrc/quantization/machete/generated/*.cu")
-  list(APPEND VLLM_EXT_SRC ${MACHETE_GEN_SOURCES})
-  message(STATUS "Machete generated sources: ${MACHETE_GEN_SOURCES}")
-
-  # See comment above for scaled_mm_c3x (same if condition)
+  # The machete kernels only work on hopper and require CUDA 12.0 or later.
   if(${CMAKE_CUDA_COMPILER_VERSION} VERSION_GREATER 12.0)
+    #
+    # For the Machete kernels we automatically generate sources for various 
+    # preselected input type pairs and schedules.
+    # Generate sources:
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} -E env 
+      PYTHONPATH=${CMAKE_CURRENT_SOURCE_DIR}/csrc/cutlass_extensions/:${CUTLASS_DIR}/python/:${VLLM_PYTHON_PATH}:$PYTHONPATH 
+        ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/csrc/quantization/machete/generate.py
+      RESULT_VARIABLE machete_generation_result
+      OUTPUT_VARIABLE machete_generation_output
+      OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/machete_generation.log
+      ERROR_FILE ${CMAKE_CURRENT_BINARY_DIR}/machete_generation.log
+    )
+
+    if (NOT machete_generation_result EQUAL 0)
+      message(FATAL_ERROR "Machete generation failed."
+                          " Result: \"${machete_generation_result}\"" 
+                          "\nCheck the log for details: "
+                          "${CMAKE_CURRENT_BINARY_DIR}/machete_generation.log")
+    else()
+      message(STATUS "Machete generation completed successfully.")
+    endif()
+
+    # Add machete generated sources
+    file(GLOB MACHETE_GEN_SOURCES "csrc/quantization/machete/generated/*.cu")
+    list(APPEND VLLM_EXT_SRC ${MACHETE_GEN_SOURCES})
+    message(STATUS "Machete generated sources: ${MACHETE_GEN_SOURCES}")
+
     set_source_files_properties(
           ${MACHETE_GEN_SOURCES}
           PROPERTIES
@@ -264,7 +267,9 @@ if(VLLM_GPU_LANG STREQUAL "CUDA")
           "-gencode arch=compute_90a,code=sm_90a")
   endif()
 
-  # Add pytorch binding
+  # Add pytorch binding for machete (add on even CUDA < 12.0 so that we can
+  #  raise an error if the user that this was built with an incompatible 
+  #  CUDA version)
   list(APPEND VLLM_EXT_SRC
     csrc/quantization/machete/machete_pytorch.cu)
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,9 @@ message(STATUS "Target device: ${VLLM_TARGET_DEVICE}")
 
 include(${CMAKE_CURRENT_LIST_DIR}/cmake/utils.cmake)
 
+# Suppress potential warnings about unused manually-specified variables
+set(ignoreMe "${VLLM_PYTHON_PATH}")
+
 #
 # Supported python versions.  These versions will be searched in order, the
 # first match will be selected.  These should be kept in sync with setup.py.

--- a/csrc/quantization/machete/machete_pytorch.cu
+++ b/csrc/quantization/machete/machete_pytorch.cu
@@ -37,7 +37,7 @@ static auto scalar_type_dispatch(ScalarType const& type, Fn fn) {
 //
 
 std::vector<std::string> supported_schedules(ScalarTypeTorchPtr const& btype) {
-#if CUDART_VERSION >= 1200
+#if defined(__CUDACC_VER_MAJOR__) && __CUDACC_VER_MAJOR__ >= 12
   return scalar_type_dispatch(*btype, [&](auto BType) {
     return GemmDispatcher<half_t, decltype(BType)>::supported_schedules();
   });
@@ -54,7 +54,7 @@ torch::Tensor gemm(torch::Tensor const& A, torch::Tensor const& B,
                    c10::optional<torch::Tensor> const& C,
                    c10::optional<double> alpha, c10::optional<double> beta,
                    c10::optional<std::string> schedule) {
-#if CUDART_VERSION >= 1200
+#if defined(__CUDACC_VER_MAJOR__) && __CUDACC_VER_MAJOR__ >= 12
   auto args = PyTorchArguments{.A = A,
                                .B = B,
                                .scales = scales,
@@ -79,7 +79,7 @@ torch::Tensor gemm(torch::Tensor const& A, torch::Tensor const& B,
 
 torch::Tensor prepack_B(torch::Tensor const& B,
                         ScalarTypeTorchPtr const& btype) {
-#if CUDART_VERSION >= 1200
+#if defined(__CUDACC_VER_MAJOR__) && __CUDACC_VER_MAJOR__ >= 12
   return scalar_type_dispatch(*btype, [&](auto BType) {
     return PrepackBDispatcher<half_t, decltype(BType), half_t>::dispatch(B);
   });

--- a/csrc/quantization/machete/machete_pytorch.cu
+++ b/csrc/quantization/machete/machete_pytorch.cu
@@ -37,9 +37,13 @@ static auto scalar_type_dispatch(ScalarType const& type, Fn fn) {
 //
 
 std::vector<std::string> supported_schedules(ScalarTypeTorchPtr const& btype) {
+#if CUDART_VERSION >= 1200
   return scalar_type_dispatch(*btype, [&](auto BType) {
     return GemmDispatcher<half_t, decltype(BType)>::supported_schedules();
   });
+#else
+  TORCH_CHECK(false, "Machete requires CUDA 12.0 or later");
+#endif
 }
 
 torch::Tensor gemm(torch::Tensor const& A, torch::Tensor const& B,
@@ -50,6 +54,7 @@ torch::Tensor gemm(torch::Tensor const& A, torch::Tensor const& B,
                    c10::optional<torch::Tensor> const& C,
                    c10::optional<double> alpha, c10::optional<double> beta,
                    c10::optional<std::string> schedule) {
+#if CUDART_VERSION >= 1200
   auto args = PyTorchArguments{.A = A,
                                .B = B,
                                .scales = scales,
@@ -67,13 +72,20 @@ torch::Tensor gemm(torch::Tensor const& A, torch::Tensor const& B,
           return GemmDispatcher<ComputeType, decltype(BType)>::dispatch(args);
         });
   });
+#else
+  TORCH_CHECK(false, "Machete requires CUDA 12.0 or later");
+#endif
 }
 
 torch::Tensor prepack_B(torch::Tensor const& B,
                         ScalarTypeTorchPtr const& btype) {
+#if CUDART_VERSION >= 1200
   return scalar_type_dispatch(*btype, [&](auto BType) {
     return PrepackBDispatcher<half_t, decltype(BType), half_t>::dispatch(B);
   });
+#else
+  TORCH_CHECK(false, "Machete requires CUDA 12.0 or later");
+#endif
 }
 
 };  // namespace machete

